modules/course_management/course_activator.py

```diff
@@ -0,0 +1,47 @@
+from typing import Literal, Union
+import discord
+from discord_slash.context import SlashContext
+import config
+from modules.course_management.util import sort_single_course
+from modules.error.friendly_error import FriendlyError
+from utils.utils import get_discord_obj, get_id
+
+
+async def activate_course(ctx: SlashContext, channel: discord.TextChannel):
+	"""Move a course from the inactive courses category to the active one."""
+	await __move_course(ctx, channel, "activate")
+
+
+async def deactivate_course(ctx: SlashContext, channel: discord.TextChannel):
+	"""Move a course from the active courses category to the inactive one."""
+	await __move_course(ctx, channel, "deactivate")
+
+
+async def deactivate_all_courses(ctx: SlashContext):
+	"""Move all active courses from the active courses to the inactive one."""
+	category: discord.CategoryChannel = get_discord_obj(
+		config.guild().categories, "ACTIVE_COURSES_CATEGORY"
+	)
+	for channel in category.text_channels:
+		await deactivate_course(ctx, channel)
+
+
+async def __move_course(
+	ctx: SlashContext,
+	channel: discord.TextChannel,
+	action: Union[Literal["activate"], Literal["deactivate"]],
+):
+	source_label, target_label = map(
+		lambda prefix: prefix + "ACTIVE_COURSES_CATEGORY",
+		("IN", "") if action == "activate" else ("", "IN"),
+	)
```

Comment:
This code takes time to comprehend, maybe it could be made easier to read?

```suggestion
source_label, target_label = (
	("INACTIVE_COURSES_CATEGORY", "ACTIVE_COURSES_CATEGORY")
	if action == "activate"
	else ("ACTIVE_COURSES_CATEGORY", "INACTIVE_COURSES_CATEGORY")
)
```